File: classes/parte-3-valores-tipos-y-variables/046-booleanos-y-valores-de-verdad/implementaciones/php/main.php
<?php
[$a, $b] = preg_split('/\s+/', trim(fgets(STDIN)));
$a = ((int) $a) !== 0;
$b = ((int) $b) !== 0;
$tf = fn($x) => $x ? "true" : "false";
printf("and=%s or=%s not_a=%s\n", $tf($a && $b), $tf($a || $b), $tf(!$a));

File: classes/parte-3-valores-tipos-y-variables/046-booleanos-y-valores-de-verdad/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func tf(x bool) string {
	if x {
		return "true"
	}
	return "false"
}

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	f := strings.Fields(line)
	ai, _ := strconv.Atoi(f[0])
	bi, _ := strconv.Atoi(f[1])
	a, b := ai != 0, bi != 0
	fmt.Printf("and=%s or=%s not_a=%s\n", tf(a && b), tf(a || b), tf(!a))
}
